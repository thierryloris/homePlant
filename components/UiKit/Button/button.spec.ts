import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";

import TedUiButton from "./index.vue";

describe("TedUiButton", () => {
  it("renders correctly with default props", () => {
    const wrapper = mount(TedUiButton, {
      props: {
        label: "Bouton",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders correctly with custom props", () => {
    const wrapper = mount(TedUiButton, {
      props: {
        label: "Bouton",
        font: "regular",
        color: "primary",
        variant: "contained",
        decoration: "icon",
        link: "/",
        align: "mauto",
        isTextOnLeft: true,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders correctly with primary color and contained variant", () => {
    const wrapper = mount(TedUiButton, {
      props: {
        label: "Bouton",
        color: "primary",
        variant: "contained",
        decoration: "icon",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders correctly with secondary color and contained variant", () => {
    const wrapper = mount(TedUiButton, {
      props: {
        label: "Bouton",
        color: "secondary",
        variant: "contained",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders correctly with secondary color and outlined variant", () => {
    const wrapper = mount(TedUiButton, {
      props: {
        label: "Bouton",
        color: "secondary",
        variant: "outlined",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders correctly with secondary color and shadow variant", () => {
    const wrapper = mount(TedUiButton, {
      props: {
        label: "Bouton",
        color: "secondary",
        variant: "shadow",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders correctly with secondary color and right variant", () => {
    const wrapper = mount(TedUiButton, {
      props: {
        label: "Bouton",
        color: "secondary",
        variant: "right",
        isTextOnLeft: true,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders correctly with transparent color and text variant", () => {
    const wrapper = mount(TedUiButton, {
      props: {
        label: "Bouton",
        color: "transparent",
        variant: "text",
        isTextOnLeft: true,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders correctly with transparent color, text variant, and bold font", () => {
    const wrapper = mount(TedUiButton, {
      props: {
        label: "Bouton",
        color: "transparent",
        variant: "text",
        font: "bold",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
