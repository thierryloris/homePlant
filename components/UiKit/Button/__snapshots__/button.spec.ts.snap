// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TedUiButton > renders correctly with custom props 1`] = `
"<chorelinkoptional to="/"><button class="bg-blue text-white font-satoshiregular m-auto rounded px-6 py-2 flex gap-3 items-center text-paragraph hover:bg-white hover:text-blue"><span>Bouton</span>
    <!--v-if-->
  </button></chorelinkoptional>"
`;

exports[`TedUiButton > renders correctly with default props 1`] = `
"<chorelinkoptional><button class="bg-blue text-white font-satoshiregular m-auto rounded px-6 py-2 flex gap-3 items-center text-paragraph hover:bg-white hover:text-blue">
    <!--v-if--><span>Bouton</span>
  </button></chorelinkoptional>"
`;

exports[`TedUiButton > renders correctly with primary color and contained variant 1`] = `
"<chorelinkoptional><button class="bg-blue text-white font-satoshiregular m-auto rounded px-6 py-2 flex gap-3 items-center text-paragraph hover:bg-white hover:text-blue">
    <!--v-if--><span>Bouton</span>
  </button></chorelinkoptional>"
`;

exports[`TedUiButton > renders correctly with secondary color and contained variant 1`] = `
"<chorelinkoptional><button class="bg-white text-blue font-satoshiregular m-auto rounded px-6 py-2 flex gap-3 items-center text-paragraph hover:bg-blue hover:text-white">
    <!--v-if--><span>Bouton</span>
  </button></chorelinkoptional>"
`;

exports[`TedUiButton > renders correctly with secondary color and outlined variant 1`] = `
"<chorelinkoptional><button class="bg-white text-blue font-satoshiregular m-auto rounded border border-blue px-6 py-2 flex gap-3 items-center text-paragraph hover:bg-blue hover:text-white">
    <!--v-if--><span>Bouton</span>
  </button></chorelinkoptional>"
`;

exports[`TedUiButton > renders correctly with secondary color and right variant 1`] = `
"<chorelinkoptional><button class="bg-white text-blue font-satoshiregular m-auto rounded px-6 py-2 flex gap-3 items-center text-paragraph"><span>Bouton</span>
    <!--v-if-->
  </button></chorelinkoptional>"
`;

exports[`TedUiButton > renders correctly with secondary color and shadow variant 1`] = `
"<chorelinkoptional><button class="bg-white text-blue font-satoshiregular m-auto rounded shadow-md px-6 py-2 flex gap-3 items-center text-paragraph hover:bg-blue hover:text-white">
    <!--v-if--><span>Bouton</span>
  </button></chorelinkoptional>"
`;

exports[`TedUiButton > renders correctly with transparent color and text variant 1`] = `
"<chorelinkoptional><button class="bg-transparent text-blue font-satoshiregular m-auto rounded-none p-0 flex gap-3 items-center text-paragraph"><span>Bouton</span>
    <!--v-if-->
  </button></chorelinkoptional>"
`;

exports[`TedUiButton > renders correctly with transparent color, text variant, and bold font 1`] = `
"<chorelinkoptional><button class="bg-transparent text-blue font-satoshibold m-auto rounded-none p-0 flex gap-3 items-center text-paragraph">
    <!--v-if--><span>Bouton</span>
  </button></chorelinkoptional>"
`;
